pkg/watermark/fetch/processor_manager.go

```diff
@@ -212,3 +212,52 @@ func (v *ProcessorManager) startHeatBeatWatcher() {
 
 	}
 }
+
+func (v *ProcessorManager) startTimeLineWatcher() {
+	watchCh, stopped := v.otWatcher.Watch(v.ctx)
+
+	for {
+		select {
+		case <-stopped:
+			return
+		case value := <-watchCh:
+			if value == nil {
+				continue
+			}
+			switch value.Operation() {
+			case store.KVPut:
+				retries := 1
+				p := v.GetProcessor(value.Key())
+				for p == nil && retries <= 10 {
+					// it is possible that by the time we receive the ot KVPut event,
+					// the heartbeat store hasn't added the new processor yet,
+					// so we need to set up the retry logic:
+					//   default processor heartbeat rate is 5 seconds
+					//   (1+..+10)*100=5500ms=5.5s
```

Comment:
open to better solution

Reply:
when a pod starts up, it first sends the heartbeat event before sending the OT event, right?

Reply:
Yes. There’s a small(?) chance that an OT put kv event happens when we haven’t completed processing heartbeat put kv events.

Reply:
Hmm if that pod continues putting new ot events then after some time it will eventually show up. But the first few events will be lost.